python/lsst/pipe/tasks/extended_psf.py

```diff
@@ -28,22 +28,62 @@
     "StackBrightStarsTask",
     "MeasureExtendedPsfConfig",
     "MeasureExtendedPsfTask",
+    "DetectorsInRegion",
 ]
 
 from dataclasses import dataclass
-from typing import List
 
 from lsst.afw.fits import Fits, readMetadata
 from lsst.afw.image import ImageF, MaskedImageF, MaskX
 from lsst.afw.math import StatisticsControl, statisticsStack, stringToStatisticsProperty
 from lsst.daf.base import PropertyList
 from lsst.geom import Extent2I
-from lsst.pex.config import ChoiceField, Config, ConfigurableField, DictField, Field, ListField
+from lsst.pex.config import ChoiceField, Config, ConfigurableField, Field, ListField, ConfigDictField
 from lsst.pipe.base import PipelineTaskConfig, PipelineTaskConnections, Struct, Task
 from lsst.pipe.tasks.coaddBase import subBBoxIter
 from lsst.pipe.base.connectionTypes import Input, Output
 
 
+def find_region_for_detector(det_id, detectors_focal_plane_regions):
+    """Find the focal plane region that contains a given detector.
+
+    Parameters
+    ----------
+    det_id : int
+        The detector ID.
+
+    detectors_focal_plane_regions : dict
+        A dictionary containing focal plane region names as keys, and the
+        corresponding detector IDs as the values.
+
+    Returns
+    -------
+    key: `str`
+        The name of the regio to which the given detector belongs.
+
+    Raises
+    ------
+    KeyError
+        Raised if the given detector is not included in any focal plane region.
+    """
+    for key, value_list in detectors_focal_plane_regions.items():
+        if det_id in value_list.detectors:
+            return key
+    raise KeyError(
+        "Detector %d is not included in any focal plane region.",
+        det_id,
+    )
+
+
+class DetectorsInRegion(Config):
+    """Provides a list of detectors that define a region.
+    """
+    detectors = ListField[int](
+        doc="A list containing the detectors IDs.",
+        default=None,
```

Comment:
Added `optional=True` as suggested. You're correct on your second point here.

Reply:
Wait, in `find_region_for_detector`, you check for membership without any checks (I can't tell yet if it has been checked previously before it gets there). May be this should be an empty list by default instead of `None` (in which case, you can remove the `optional=True`.

Reply:
That's a nice idea. Modified as you suggest.